data/abi/abi_encode.go

```diff
@@ -322,15 +314,31 @@ func compressBools(boolSlice []interface{}) (uint8, error) {
 	return res, nil
 }
 
+// decodeUint decodes byte slice into golang int/big.Int
+func decodeUint(encoded []byte, bitSize uint16) (interface{}, error) {
+	if len(encoded) != int(bitSize)/8 {
+		return nil,
+			fmt.Errorf("uint/ufixed decode: expected byte length %d, but got byte length %d", bitSize/8, len(encoded))
+	}
+	switch bitSize / 8 {
+	case 1:
+		return encoded[0], nil
+	case 2:
+		return uint16(new(big.Int).SetBytes(encoded).Uint64()), nil
+	case 3, 4:
+		return uint32(new(big.Int).SetBytes(encoded).Uint64()), nil
+	case 5, 6, 7, 8:
+		return new(big.Int).SetBytes(encoded).Uint64(), nil
```

Comment:
nit: `binary.BigEndian.Uint16/Uint32/Uint64(encoded)` might be better than converting to bigint first

Reply:
This is also a ugly workaround, since the `binary.BigEdian` methods enforce the byte length to be 2/4/8.

if you are attempting 65537: `[1, 0, 1]`, then you will have array index out of scope error...